core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -274,8 +274,36 @@ private[spark] class DAGScheduler(
       accumUpdates: Seq[AccumulatorV2[_, _]],
       metricPeaks: Array[Long],
       taskInfo: TaskInfo): Unit = {
+    // It may cause heavy full GC problem if some heavy external accumulators keep in memory.
+    // We update these heavy accumulators before they entering into Spark listener event loop.
+    val (heavyAccumUpdates, otherAccumUpdates) = accumUpdates.partition { acc =>
+      acc.name.exists(_.startsWith(ExternalHeavyAccumulator.HEAVY_PREFIX))
+    }
+    heavyAccumUpdates.foreach { updates =>
+      val id = updates.id
```

Comment:
This code part looks almost the same with `updateAccumulators`, so could we share it between them?

Reply:
Yes. We can extract the common part as a new private method.